components/Forums/ThreadCommentCard.js

```diff
@@ -0,0 +1,240 @@
+
+  async _handleRemoveButtonClick (event) {
+    const {
+      name,
+    } = event.target
+
+    const {
+      confirmingRemove,
+    } = this.state
+
+    const {
+      deleteThreadComment,
+      onDelete,
+      comment,
+    } = this.props
+
+    if (confirmingRemove && name === 'confirm') {
+      this.setState({ confirmingRemove: false, removing: true })
+
+      const { status } = await deleteThreadComment(comment.id)
+
+      if (status === 'success') {
+        this.setState({ removing: false, removed: true })
+
+        if (typeof onDelete === 'function') {
+          onDelete(comment)
+        }
+      } else {
+        this.setState({ removing: false })
+      }
+    } else if (confirmingRemove && name === 'deny') {
+      this.setState({ confirmingRemove: false })
+    } else {
+      this.setState({ confirmingRemove: true })
+    }
+  }
+
+  /***************************************************************************\
+    Public Methods
+  \***************************************************************************/
+
+  async componentDidMount () {
+    let { user } = this.props
+
+    if (!user) {
+      const { payload, status } = await this.props.getUser(this.props.posterId)
+
+      if (status === 'success') {
+        user = { ...payload.data }
+      }
+    }
+
+    this.setState({ user })
+  }
+
+  constructor (props) {
+    super(props)
+
+    this._bindMethods([
+      '_handleRemoveButtonClick',
+    ])
+
+    this.state = {
+      user: null,
+      removing: false,
+      removed: false,
+    }
+  }
+
+
+  renderDeleteButtons () {
+    const {
+      confirmingRemove,
+      removing,
+      removed,
+    } = this.state
+
+    return (
+      <React.Fragment>
+        {confirmingRemove && (
+          <React.Fragment>
+            <span>Are you sure?</span>
+            <button
+              className="danger"
+              name="deny"
+              onClick={this._handleRemoveButtonClick}>
+              No
+            </button>
+          </React.Fragment>
+        )}
+        <button
+          className={confirmingRemove ? 'success' : 'danger'}
+          name="confirm"
+          disabled={removing || removed}
+          onClick={this._handleRemoveButtonClick}>
+          {!removing && !confirmingRemove && (removed ? 'This doesn\'t do anything yet!' : 'Delete')}
+
+          {!removing && confirmingRemove && 'Yes'}
+
+          {removing && (
+            <span name="confirm"><FontAwesomeIcon icon="spinner" pulse /> Deleting...</span>
+          )}
+        </button>
+      </React.Fragment>
+    )
+  }
+
+  render () {
+    const {
+      currentUserIsPoster,
+      comment,
+      id,
+    } = this.props
+
+    const {
+      user,
+      removed,
+    } = this.state
+
+    /* eslint-disable camelcase  */
+    const {
+      inserted_at,
+    } = comment.attributes
+    /* eslint-enable camelcase  */
+
+    const instertedAtMoment = moment.utc(inserted_at)
```

Comment:
We could avoid having to disable eslint with this:

```
const instertedAtMoment = moment.utc(comment.attributes.inserted_at)
```

Reply:
Ah yeah. The destructure was back when I was using inserted_at in multiple locations. Rather unneeded now that everything is done through the moment